R/best_and_worst_cases.R

```diff
@@ -0,0 +1,23 @@
+best_and_worst_cases <- function(data) {
+  data |>
+    mutate(amount_of_distinct_products = dplyr::n_distinct(.data$ep_product), .by = "companies_id") |>
```

Comment:
I see "amount", "n", and "count" but they all seem to mean the same thing. I prefer "n" because it's standard, clear, and short.

Reply:
It's best to avoid hard-coding brittle column names like `.data$ep_product`, and `by = "companies_id"`. 

We (developers) don't control those names, and when they change it's best to have them all separate in a single file. The SOLID principle behind this idea is "Single responsibility":

> Gather together those things that change for the same reason, and separate those things that change for different reasons.

We have implemented this idea in a few places but the simplest one in [tiltAddCO2](https://github.com/2DegreesInvesting/tiltAddCO2/blob/main/R/add_co2-column-names.R).

Reply:
BTW here I'm trying to embrace my own advice

https://github.com/2DegreesInvesting/tiltAddCO2/pull/2

Reply:
Adapted these changes as suggested :)